### PR TITLE
tidy

### PR DESCRIPTION
**Pull request by cdmistman**
Why
===

i had a laundry list of things to do before writing and sharing the blog post. this is a mega-PR for achieving this.

What changed
============

i suggest looking at each commit individually. i'll be avoiding force-pushes to ensure review comments on specific commits aren't lost. overall list of changes:
1. define a common type to use between both CLIs for the database representation of packages
2. improve error reporting
3. remove system argument from the genRegistry nix function
4. `rippkgs-index` now uses `nix eval` with the `genRegistry` function to generate the registry (so only 1 registry format is needed now)
5. refactored both CLIs for clarity
6. updated and added info to the README

Test plan
=========

1. `cargo build --release --all`
2. follow the readme
3. it works

### CODE REVIEW

**Comment by airportyh** [comment]
Oh, `nix develop`...

File: src/bin/search/main.rs [hunk 1]
```diff
@@ -89,24 +89,24 @@ fn main() -> Result<()> {
...
             opts.filter_built,
         )
-        .context("error searching for fuzzy query")?;
-        serde_json::to_value(results).context("error serializing fuzzy results")?
+        .context("searching for fuzzy query")?;
+        serde_json::to_value(results).context("serializing fuzzy results")?
     };
 
-    serde_json::to_writer(stdout(), &results).context("error printing results")?;
+    serde_json::to_writer(stdout(), &results).context("printing results")?;
```
Comment (airportyh) [inline src/bin/search/main.rs:103]: It's customary to not say "error" in front? Will the handler automatically add verbiage to that effect?

File: lib/default.nix [hunk 2]
```diff
@@ -33,7 +33,7 @@ in {
         };
       };
 
-      platformForAvailability = {system = platform;};
+      platformForAvailability = {system = pkgs.system or builtins.currentSystem;};
```
Comment (airportyh) [inline lib/default.nix:36]: Allowing impure?

File: src/bin/search/main.rs (hunk 1 above)
Comment (cdmistman) [reply src/bin/search/main.rs:103]: yes!
image

File: src/bin/index/main.rs [hunk 3]
```diff
@@ -125,30 +125,36 @@ fn get_registry(
...
+        let nixpkgs_include_arg = format!("nixpkgs={nixpkgs}");
+        let nixpkgs_config = nixpkgs_config
+            .as_ref()
+            .map(|config| config.display().to_string())
+            .unwrap_or("{}".to_string());
+        let apply_arg = format!(
+            r#"
+genRegistry:
+
+let pkgs = import <nixpkgs> {{ config = {nixpkgs_config}; }};
```
[the remark this answers is not in the document]
Comment (cdmistman) [reply src/bin/index/main.rs]: we'll fix this in a future PR

**Pull request merged**